[docs][glossary] Add page that lists all standardized k8s terms

abiogenesis-now opened the pull request:
@chenopis This PR is a quick prototype of what we discussed last week. You mentioned that Sarah (or was it Paris?) thought it would be worth formally defining "user persona terms" like "cluster operator" and "application developer."

I put the link to this glossary page under *Reference*. You may have mentioned putting it on the home page--I think that categorically it makes sense under "Reference", but was that suggestion due to visibility concerns? If you feel strongly I can move it.

**GIF of final product, illustrating all the random features:**

![glossary](https://user-images.githubusercontent.com/25401650/31300345-15f0beea-aaa8-11e7-959b-d0a3e23bed81.gif)


<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/5657)
<!-- Reviewable:end -->

abiogenesis-now commented:
See https://deploy-preview-5657--kubernetes-io-master-staging.netlify.com/docs/reference/glossary/

chenopis commented:
cc @parispittman

chenopis commented:
@abiogenesis-now It was Paris (@parispittman) who mentioned adding community terms, but I like to think of it as coming from the uber-entity known as Saris. LOL

abiogenesis-now commented:
@chenopis addressed your comments! Do you think we can merge this first or should we keep it as a PR until we've fleshed out all the "People"/"User Persona" terms?

chenopis commented:
@abiogenesis-now I want to merge this now, but since it's still pretty sparse, I think we should do a quick sprint to get maybe a dozen terms in before we merge.

abiogenesis-now commented:
@chenopis good idea! I'll make a list--might solicit help from you, Luke, and Jennifer to help populate/edit the definitions (if you have bandwidth, of course!)

chenopis commented:
We can also scrape some terminology and/or definitions from: https://github.com/kubernetes-incubator/service-catalog/blob/master/terminology.md

abiogenesis-now commented:
Got carried away so I updated the glossary UI to be fancy and tag-filtering.
It *might* have bugs and not scale with a large number of terms (because essentially ALL content is being loaded onto the page and I'm using Javascript to hide/show things accordingly...maybe JS will be able to though, not sure).

TODOs:
* add Paris's terms to the YAML
* flesh out definitions
* make sure tags make sense (and how we assigned things)
* fix the CI test that checks canonical terms (since I changed the structure of those)
* current setup kind of assumes that each term has at least one tag---I'll add that check to the CI test

chenopis commented:
FYI, Travis CI is failing

> [0K$ go test -v k8s.io/kubernetes.github.io/test
=== RUN   TestExampleObjectSchemas
--- PASS: TestExampleObjectSchemas (0.10s)
=== RUN   TestReadme
--- PASS: TestReadme (0.01s)
=== RUN   TestCanonicalTags
--- FAIL: TestCanonicalTags (0.00s)
	glossary_test.go:43: Unable to read file ../_data/canonical-tags.yml: open ../_data/canonical-tags.yml: no such file or directory
FAIL
exit status 1
FAIL	k8s.io/kubernetes.github.io/test	0.182s

chenopis commented:
@abiogenesis-now For some reason, the "Create an Issue" button seems to have moved.

![51hhvz4oq1b](https://user-images.githubusercontent.com/24798125/31245434-2f68368a-a9bf-11e7-9d55-b50dfff20b29.png)

abiogenesis-now commented:
@chenopis good call, I'll add it to the TODOS

abiogenesis-now commented:
Terms remaining:

- [ ] application architect
- [ ] application developer
- [ ] cluster
- [ ] container
- [ ] CronJob
- [ ] deployment
- [ ] downstream
- [ ] helm chart
- [ ] ingress
- [ ] istio
- [ ] kops
- [ ] kubeadm
- [ ] kubectl
- [ ] Kubernetes API
- [ ] minikube
- [ ] RBAC
- [ ] service
- [ ] upstream

chenopis commented:
@abiogenesis-now Can we change the spec and rename "formerly" to "aka"?

chenopis commented:
@bgrant0607 @matchstick @thockin @parispittman We are relaunching Glossary w/ a starter deck of 32 terms and will add more over time. The idea behind this framework is that we can just drop new terms into the `/_data/glossary/` directory and they will automatically show up in the new Glossary page. If you have a moment, please look through the definitions in the [preview](https://deploy-preview-5657--kubernetes-io-master-staging.netlify.com/docs/reference/glossary/) and add review comments if anything doesn't look quite right. We would like to merge this PR ASAP. Thx!

abiogenesis-now commented:
Added GIF to main description (for reference)

chenopis commented:
cc @jaredbhatti @devin-donnelly

thockin commented on _data/glossary/application-developer.yaml:
```diff
@@ -0,0 +1,10 @@
+short-description: |
+  A person who writes an application that runs in a Kubernetes cluster.
+long-description: |
+  Typically an application developer works on one portion of an application that is split into microservices.
```
I don't think we need to say that microsrvices are "typical".  Maybe "..one portion of an application, which might be monolithic or composed of many microservices."

chenopis replied:
Ok, I'll make the change.

thockin commented on _data/glossary/cluster-architect.yaml:
```diff
@@ -0,0 +1,12 @@
+  A person who designs infrastructure that involves one or more Kubernetes clusters.
+
+long-description: |
+  They are concerned with distributed system best practices, such as high availability and security. A cluster architect is likely the one determining the suitability of a feature like cross-cloud cluster federation, for their particular setup.
```
I would strike references to federation.

chenopis replied:
Ok, I'll remove the second sentence.

thockin commented on _data/glossary/cluster.yaml:
```diff
@@ -0,0 +1,9 @@
+short-description: |
+  A set of machines, called nodes, that run containerized applications managed by Kubernetes.
+long-description: |
+  Typically a cluster has serveral worker nodes and at least one master node.
```
typo: serveral

chenopis replied:
got it, thx

thockin commented on _data/glossary/code-contributor.yaml:
```diff
@@ -0,0 +1,11 @@
+short-description: |
+  Develops and contributes code to the project that is part of the open source codebase.
+long-description: |
+  They are also an active community member and participate in one of the Special Interest Groups (SIGs).
```
"may participate"

"one or more"

chenopis replied:
thx

thockin commented on _data/glossary/cronjob.yaml:
```diff
@@ -0,0 +1,9 @@
+- core-object
+- workload
+short-description: |
+  Manages time based [Jobs](/docs/concepts/jobs/run-to-completion-finite-workloads/) that are run once or periodically on a given schedule.
```
"once" doesn't make sense "on a given schedule".

"A job that runs on a periodic schedule. "?

chenopis replied:
Ok, I'm going with "Manages a [Job](/docs/concepts/jobs/run-to-completion-finite-workloads/) that runs on a periodic schedule."

thockin commented on _data/glossary/deployment.yaml:
```diff
@@ -0,0 +1,10 @@
+short-description: |
+  An API object that manages a replicated application.
+long-description: |
+  Each replica is represented by a Pod, and the Pods are distrubuted among the Nodes of a cluster.
```
typo: distrubuted

chenopis replied:
thx

thockin commented on _data/glossary/ingress.yaml:
```diff
@@ -0,0 +1,10 @@
+- architecture
+- extension
+short-description: |
+  An API object that manages external access to the services in a cluster.
```
cluster, typically HTTP.

chenopis replied:
got it

thockin commented on _data/glossary/kubernetes-api.yaml:
```diff
@@ -0,0 +1,10 @@
+- fundamental
+- architecture
+short-description: |
+  The foundation for how Kubernetes represents and stores the state of the cluster.
```
This doesn't actually describe what an apiserver is.

"The application which serves the Kubernetes HTTP API, and which stores the state of the cluster".

chenopis replied:
I'm going with "The application which serves the Kubernetes HTTP API and stores the state of the cluster."

thockin commented on _data/glossary/maintainer.yaml:
```diff
@@ -0,0 +1,8 @@
+tags:
+- community
+short-description: |
+  Works holistically across the project to maintain its health and success.
```
This is what they do, not what they are.

chenopis replied:
@abiogenesis-now Let's rewrite this.

chenopis replied:
How about...

**short-description:** "A highly experienced contributor, active in multiple areas, who has cross area ownership and GitHub repo write access."

**long-description:** "They work holistically across the project to maintain its health and success and have made substantial contributions, both through code development and broader organizational efforts."

abiogenesis-now replied:
works for me---I'm making the slight copyedit of saying "active in multiple areas [of Kubernetes]" but if you disagree, let me know

thockin commented on _data/glossary/pod.yaml:
```diff
@@ -0,0 +1,14 @@
+- core-object
+- fundamental
+short-description: |
+  The smallest and simplest Kubernetes object. Represents a set of running processes on your cluster.
```
s/processes/containers ?

chenopis replied:
ok

thockin commented on _data/glossary/service.yaml:
```diff
@@ -0,0 +1,9 @@
+- fundamental
+- core-object
+short-description: |
+  An API object that manages load-balanced access to a replicated set of Pods.
```
An API object that describes how to access applications, such as a set of Pods, and can describe ports and load-balancers.

chenopis replied:
thx

chenopis commented:
BTW, we should double check the community membership role definitions against https://github.com/kubernetes/community/blob/master/community-membership.md.

zacharysarah commented on _data/canonical-tags/fundamental.yaml:
```diff
@@ -0,0 +1,3 @@
+id: fundamental
+name: Fundamental
+description: Relevant to a first-time user of Kubernetes.
```
Relevant for

zacharysarah commented on _data/canonical-tags/networking.yaml:
```diff
@@ -0,0 +1,3 @@
+id: networking
+name: Networking
+description: Relevant to how Kubernetes components talk to each other (and to programs outside the cluster).
```
> How Kubernetes components talk to each other and to programs outside the cluster.

abiogenesis-now replied:
:+1: I'll be changing all of these as you suggested, but want to provide some context. Initially I prefaced a lot of these terms with "relevant to" because the description shows up as follows:
<img width="1092" alt="screen shot 2017-10-10 at 8 57 42 am" src="https://user-images.githubusercontent.com/25401650/31396742-1e3ef050-ad99-11e7-85f0-be055ceec424.png">

I wasn't sure if the description should directly apply to the tags themselves (in which case, it's clear that "relevant for" should be omitted), or to the *terms* described by the tags---so I initially opted for the latter.

zacharysarah replied:
Thanks for the context, it helps! 👍

zacharysarah commented on _data/canonical-tags/architecture.yaml:
```diff
@@ -0,0 +1,3 @@
+id: architecture
+name: Architecture
+description: Related to the inner components of the Kubernetes system.
```
> the Kubernetes system

Can we replace this with just "Kubernetes"? See `operation.yaml` for the same question. So:

> description: The inner components of Kubernetes.

zacharysarah commented on _data/canonical-tags/operation.yaml:
```diff
@@ -0,0 +1,3 @@
+id: operation
+name: Operation
+description: Relevant to starting and maintaining the Kubernetes system itself.
```
> the Kubernetes system

Can we replace this with "Kubernetes"? See `architecture.yaml` for the same question. So:

> description: Starting and maintaining Kubernetes.

zacharysarah commented on _data/canonical-tags/security.yaml:
```diff
@@ -0,0 +1,3 @@
+id: security
+name: Security
+description: Relevant to keeping Kubernetes applications safe and secure.
```
description: Keeping Kubernetes applications safe and secure.

zacharysarah commented on _data/canonical-tags/storage.yaml:
```diff
@@ -0,0 +1,3 @@
+id: storage
+name: Storage
+description: Relevant to how Kubernetes applications persist data.
```
description: How Kubernetes applications handle persistent data.

zacharysarah commented on _data/canonical-tags/extension.yaml:
```diff
@@ -0,0 +1,3 @@
+id: extension
+name: Extension
+description: Related to the supported customization of Kubernetes.
```
description: Supported customizations of Kubernetes

zacharysarah commented on _data/canonical-tags/community.yaml:
```diff
@@ -0,0 +1,3 @@
+id: community
+name: Community
+description: Related to Kubernetes open-source development.
```
I was originally going to change the hell out of this, but the more I think about it, the more I think this works.

zacharysarah commented on _data/canonical-tags/tool.yaml:
```diff
@@ -0,0 +1,3 @@
+id: tool
+name: Tool
+description: Separate software that makes Kubernetes easier or better to use.
```
description: Software that makes Kubernetes easier or better to use.

zacharysarah commented on _data/canonical-tags/workload.yaml:
```diff
@@ -0,0 +1,3 @@
+id: workload
+name: Workload
+description: Relevant to running applications ON Kubernetes.
```
description: Applications running on Kubernetes.

zacharysarah commented on _data/glossary/_example.yml:
```diff
@@ -1,13 +1,13 @@
 id: _example
 name: Example K8s Term
-formerly:
+aka:
```
Say more about using "aka" here. Would it be better to provide an example that matches other glossary entries, and use "related", for example?

abiogenesis-now replied:
Hm, could you clarify what you mean by "an example that matches other glossary entries"? Thanks!

chenopis replied:
I suggested aka (also known as). We could add an explanation of the abbreviation in the example file.

zacharysarah replied:
@chenopis I think we're good as is--I'm probably overthinking it.

zacharysarah commented on _data/glossary/application-architect.yaml:
```diff
@@ -0,0 +1,11 @@
+tags:
+  - user-type
+short-description: |
+  A person concerned with the high-level design of an application.
```
Replace "concerned with" with "responsible for".

zacharysarah commented on _data/glossary/application-architect.yaml:
```diff
@@ -0,0 +1,11 @@
+short-description: |
+  A person concerned with the high-level design of an application.
+long-description: |
+  The architect typically ensures that the app's implementation allows it to interact with its surrounding components in a scalable, maintainable way.
```
An architect ensures that an app's implementation ...

zacharysarah commented on _data/glossary/application-architect.yaml:
```diff
@@ -0,0 +1,11 @@
+  A person concerned with the high-level design of an application.
+long-description: |
+  The architect typically ensures that the app's implementation allows it to interact with its surrounding components in a scalable, maintainable way.
+  These other components are often databases, logging infrastructure, and other microservices.
```
Surrounding components include databases, ...

zacharysarah commented on _data/glossary/application-developer.yaml:
```diff
@@ -0,0 +1,10 @@
+short-description: |
+  A person who writes an application that runs in a Kubernetes cluster.
+long-description: |
+  Typically an application developer works on one portion of an application, which might be monolithic or composed of many microservices.
```
An application developer focuses on one part of an application. The scale of their focus may vary significantly in size.

zacharysarah commented on _data/glossary/approver.yaml:
```diff
@@ -0,0 +1,10 @@
+tags:
+- community
+short-description: |
+  Able to both review and approve K8s code contributions.
```
A person who can review and approve Kubernetes code contributions.

zacharysarah commented on _data/glossary/approver.yaml:
```diff
@@ -0,0 +1,10 @@
+short-description: |
+  Able to both review and approve K8s code contributions.
+long-description: |
+  While code review is focused on code quality and correctness, approval is focused on holistic acceptance of a contribution including&#58; backwards/forwards compatibility, adhering to API and flag conventions, subtle performance and correctness issues, interactions with other parts of the system, etc.
```
While code review focuses on code quality and correctness, approval is focused on holistic acceptance of a contribution. Holistic acceptance includes backwards/forwards compatibility, adhering to API and flag conventions, subtle performance and correctness issues, interactions with other parts of the system, and others.

zacharysarah commented on _data/glossary/cla.yaml:
```diff
@@ -0,0 +1,8 @@
+tags:
+- community
+short-description: |
+  Terms under which the contributor grants a license to the open source project for their contributions.
```
Terms under which a contributor grants a license to an open source project for their contributions.

zacharysarah commented on _data/glossary/cla.yaml:
```diff
@@ -0,0 +1,8 @@
+short-description: |
+  Terms under which the contributor grants a license to the open source project for their contributions.
+long-description: |
+  They help enable a legal resolution to cases in which there are copyright disputes related to contributed material or intellectual property (IP).
```
CLAs help resolve legal disputes involving contributed material and intellectual property.

zacharysarah commented on _data/glossary/cluster-architect.yaml:
```diff
@@ -0,0 +1,12 @@
+  A person who designs infrastructure that involves one or more Kubernetes clusters.
+
+long-description: |
+  They are concerned with distributed system best practices, such as high availability and security.
```
Cluster architects are concerned with best practices for distributed systems, for example: high availability and security.

zacharysarah commented on _data/glossary/kops.yaml:
```diff
@@ -0,0 +1,18 @@
+  A CLI tool that helps you create, destroy, upgrade and maintain production-grade, highly available, Kubernetes clusters. *Officially supports AWS only, with GCE and VMware vSphere in alpha*.
+long-description: |
+  `kops` provisions your cluster with the following opinionated approach&#58;
+
```
Remove blank line.

zacharysarah commented on _data/glossary/kops.yaml:
```diff
@@ -0,0 +1,18 @@
+short-description: |
+  A CLI tool that helps you create, destroy, upgrade and maintain production-grade, highly available, Kubernetes clusters. *Officially supports AWS only, with GCE and VMware vSphere in alpha*.
+long-description: |
+  `kops` provisions your cluster with the following opinionated approach&#58;
```
```
`kops` provisions your cluster with:
```

zacharysarah commented on _data/glossary/kops.yaml:
```diff
@@ -0,0 +1,18 @@
+    * DNS-based cluster identification
+    * Self-healing&#58; everything runs in Auto-Scaling Groups
+    * Limited OS support (Debian preferred, Ubuntu 16.04 supported, early support for CentOS & RHEL)
+    * High-Availability support
```
* High availability support

zacharysarah commented on _data/glossary/kops.yaml:
```diff
@@ -0,0 +1,18 @@
+    * High-Availability support
+    * The ability to directly provision, or generate terraform manifests
+
+  If your opinions differ from these you may prefer to build your own cluster using `kubeadm` as a building block. `kops` builds on the kubeadm work.
```
```
You can also build your own cluster using `kubeadm` as a building block. `kops` builds on the kubeadm work.
```

zacharysarah commented on _data/glossary/kubeadm.yaml:
```diff
@@ -0,0 +1,9 @@
+- tool
+- operation
+short-description: |
+  A tool that you can use to quickly install Kubernetes and set up a secure cluster.
```
A tool for quickly installing Kubernetes and setting up a secure cluster.

zacharysarah commented on _data/glossary/kubeadm.yaml:
```diff
@@ -0,0 +1,9 @@
+short-description: |
+  A tool that you can use to quickly install Kubernetes and set up a secure cluster.
+long-description: |
+  You can use it to install both the control plane and the worker node components.
```
You can use kubeadm

zacharysarah commented on _data/glossary/kubectl.yaml:
```diff
@@ -0,0 +1,9 @@
+- tool
+- fundamental
+short-description: |
+  A command line tool that you can use to communicate with a Kubernetes API server.
```
A command line tool for communicating with a Kubernetes API server.

zacharysarah commented on _data/glossary/kubectl.yaml:
```diff
@@ -0,0 +1,9 @@
+short-description: |
+  A command line tool that you can use to communicate with a Kubernetes API server.
+long-description: |
+  You can use kubectl to create, inspect, update, and delete Kubernetes objects.
```
👍

zacharysarah commented on _data/glossary/kubernetes-api.yaml:
```diff
@@ -0,0 +1,10 @@
+- fundamental
+- architecture
+short-description: |
+  The application which serves the Kubernetes HTTP API and stores the state of the cluster.
```
"The application that serves Kubernetes functionality through a RESTful interface and stores the state of the cluster."

I'm open to a different description, but I'd prefer to avoid the tautology of defining an API as an API. 😎

abiogenesis-now replied:
makes sense! :P

zacharysarah commented on _data/glossary/kubernetes-api.yaml:
```diff
@@ -0,0 +1,10 @@
+short-description: |
+  The application which serves the Kubernetes HTTP API and stores the state of the cluster.
+long-description: |
+  Kubernetes resources and "records of intent" are all stored as API objects, and modified via REST calls to the API. The API allows configuration to be managed in a declarative way.
```
via RESTful calls

zacharysarah commented on _data/glossary/kubernetes-api.yaml:
```diff
@@ -0,0 +1,10 @@
+  The application which serves the Kubernetes HTTP API and stores the state of the cluster.
+long-description: |
+  Kubernetes resources and "records of intent" are all stored as API objects, and modified via REST calls to the API. The API allows configuration to be managed in a declarative way.
+  Users can interact with the Kubernetes API directly, or via tools like `kubectl`. The core Kubernetes API if flexible and can also be extended to support custom resources.
```
The core Kubernetes API is flexible

zacharysarah commented on _data/glossary/maintainer.yaml:
```diff
@@ -0,0 +1,8 @@
+tags:
+- community
+short-description: |
+  A highly experienced contributor, active in multiple areas of Kubernetes, who has cross-area ownership and GitHub repo write access.
```
and write access to a project's GitHub repository.

zacharysarah commented on _data/glossary/maintainer.yaml:
```diff
@@ -0,0 +1,8 @@
+short-description: |
+  A highly experienced contributor, active in multiple areas of Kubernetes, who has cross-area ownership and GitHub repo write access.
+long-description: |
+  They work holistically across the project to maintain its health and success and have made substantial contributions, both through code development and broader organizational efforts.
```
Maintainers work

zacharysarah commented on _data/glossary/member.yaml:
```diff
@@ -0,0 +1,8 @@
+short-description: |
+  A continuously active contributor in the K8s community.
+long-description: |
+  They can have issues and PRs assigned to them, participate in SIGs through GitHub teams, and pre-submit tests are automatically run for their PRs. A member is expected to remain an active contributor to the community.
```
Members can have issues and PRs assigned to them and participate in special interest groups (SIGs) through GitHub teams. Pre-submit tests are automatically run for members' PRs. A member is expected to remain an active contributor to the community.

zacharysarah commented on _data/glossary/minikube.yaml:
```diff
@@ -0,0 +1,9 @@
+- fundamental
+- tool
+short-description: |
+  A tool that makes it easy to run Kubernetes locally.
```
A tool for running Kubernetes locally.

zacharysarah commented on _data/glossary/pod.yaml:
```diff
@@ -0,0 +1,14 @@
+- core-object
+- fundamental
+short-description: |
+  The smallest and simplest Kubernetes object. Represents a set of running containers on your cluster.
```
A pod represents a set of running containers on your cluster.

zacharysarah commented on _data/glossary/pod.yaml:
```diff
@@ -0,0 +1,14 @@
+short-description: |
+  The smallest and simplest Kubernetes object. Represents a set of running containers on your cluster.
+long-description: |
+  A pod is typically set up to run a single primary container. It can also run optional sidecar containers that add supplementary features like logging. Pods are commonly managed by a Deployment.
```
Maybe link to "Deployment"?

zacharysarah commented on _data/glossary/reviewer.yaml:
```diff
@@ -0,0 +1,10 @@
+tags:
+- community
+short-description: |
+  Able to review code for quality and correctness on some part of the project.
```
A person who reviews code for quality and correctness on some part of the project.

zacharysarah commented on _data/glossary/reviewer.yaml:
```diff
@@ -0,0 +1,10 @@
+short-description: |
+  Able to review code for quality and correctness on some part of the project.
+long-description: |
+  They are knowledgeable about both the codebase and software engineering principles.
```
Reviewers are knowledgeable

zacharysarah commented on _data/glossary/reviewer.yaml:
```diff
@@ -0,0 +1,10 @@
+long-description: |
+  They are knowledgeable about both the codebase and software engineering principles.
+
+  *Reviewer status is scoped to a part of the codebase.*
```
Remove italics and join to previous line.

zacharysarah commented on _data/glossary/reviewer.yaml:
```diff
@@ -0,0 +1,10 @@
+  Able to review code for quality and correctness on some part of the project.
+long-description: |
+  They are knowledgeable about both the codebase and software engineering principles.
+
```
Delete blank line.

zacharysarah commented on _data/glossary/service.yaml:
```diff
@@ -0,0 +1,9 @@
+- fundamental
+- core-object
+short-description: |
+  An API object that describes how to access applications, such as a set of Pods, and can describe ports and load-balancers.
```
Consistent capitalization: Pods/pods

zacharysarah commented on _data/glossary/sig.yaml:
```diff
@@ -0,0 +1,10 @@
+id: sig
+name: SIG (Special Interest Group)
```
special interest group (no caps)

zacharysarah commented on _data/glossary/sig.yaml:
```diff
@@ -0,0 +1,10 @@
+tags:
+- community
+short-description: |
+  Manages an ongoing subproject or aspect of the larger Kubernetes open source project.
```
Members who collectively manage an ongoing piece or aspect of the larger Kubernetes open source project.

zacharysarah commented on _data/glossary/sig.yaml:
```diff
@@ -0,0 +1,10 @@
+short-description: |
+  Manages an ongoing subproject or aspect of the larger Kubernetes open source project.
+long-description: |
+  Members within a SIG have a shared interest in advancing a specific area, such as architecture, API machinery, documentation, etc.
```
API machinery, or documentation.

zacharysarah commented on _data/glossary/sig.yaml:
```diff
@@ -0,0 +1,10 @@
+  Manages an ongoing subproject or aspect of the larger Kubernetes open source project.
+long-description: |
+  Members within a SIG have a shared interest in advancing a specific area, such as architecture, API machinery, documentation, etc.
+  They must follow the [SIG Governance](https://github.com/kubernetes/community/blob/master/sig-governance.md) guidelines but can have their own contribution policy and channels of communication.
```
SIGs must follow

zacharysarah commented on _data/glossary/statefulset.yml:
```diff
@@ -1,18 +1,18 @@
+- storage
 short-description: |
-  Manage the deployment and scaling of a set of Pods, *and provide guarantees about ordering*. They do so by maintaining a *unique*, sticky identity for each of their Pods.
+  Manages the deployment and scaling of a set of Pods, *and provides guarantees about the ordering and uniqueness* of these Pods.
```
1. Remove italics.
2. Consistent caps: Pods/pods

abiogenesis-now replied:
1. I think this is worth italicizing because this is chiefly what distinguishes StatefulSets from Deployments. Let me know if you strongly want to change it though.
2. Going to consistently capitalize as Pods (to distinguish it as a K8s object).

zacharysarah commented on _data/glossary/statefulset.yml:
```diff
@@ -1,18 +1,18 @@
+  Like a Deployment, a StatefulSet manages Pods that are based on an identical container spec. Unlike a Deployment, a StatefulSet maintains a sticky identity for each of their Pods. These pods are created from the same spec, but are not interchangeable--each has a persistent identifier that it maintains across any rescheduling.
 
-  StatefulSets also operate according to the Controller pattern. You define your desired state in a StatefulSet *object*, and the StatefulSet *controller* makes any necessary updates to the get there from the current state.
+  A StatefulSet operates under the same pattern as any other Controller. You define your desired state in a StatefulSet *object*, and the StatefulSet *controller* makes any necessary updates to get there from the current state.
```
Consistent caps: Controller/controller

abiogenesis-now replied:
I feel like this capitalization denotes Controller as something special in K8s world, as opposed to some sort of other meaning of "controller" (e.g. MVC controller)?

zacharysarah replied:
That's real. I think "Controller" works here because it distinguishes not strictly from other controllers (e.g. MVC) in general, but among other K8s Controllers as well. (e.g., Admission, Replication)

zacharysarah commented on _data/glossary/statefulset.yml:
```diff
@@ -1,18 +1,18 @@
 
 long-description: |
-  Like Deployments, StatefulSets manage Pods that are based on an identical container spec. However, although their specs are the same, the Pods in a StatefulSet are not interchangeable. Each Pod has a persistent identifier that it maintains across any rescheduling.
+  Like a Deployment, a StatefulSet manages Pods that are based on an identical container spec. Unlike a Deployment, a StatefulSet maintains a sticky identity for each of their Pods. These pods are created from the same spec, but are not interchangeable--each has a persistent identifier that it maintains across any rescheduling.
```
interchangeable: each has

zacharysarah commented on _data/glossary/upstream.yaml:
```diff
@@ -0,0 +1,12 @@
+short-description: |
+  May refer to: core Kubernetes or the source repo from which a repo was forked.
+long-description: |
+  In the Kubernetes Community, conversations often use *upstream* to mean the core Kubernetes codebase, which the general ecosystem, other code, or third-party tools rely upon. For example, community members may suggest that a feature is moved upstream so that it is in the core codebase instead of in a plugin or third-party tool.
```
relies upon.

zacharysarah commented on _data/glossary/wg.yaml:
```diff
@@ -0,0 +1,9 @@
+id: wg
+name: WG (Working Group)
```
working group

zacharysarah commented on _data/glossary/wg.yaml:
```diff
@@ -0,0 +1,9 @@
+short-description: |
+  Facilitates the discussion and/or implementation of a short-lived, narrow, or decoupled project for a committee, SIG, or cross-SIG effort.
+long-description: |
+  They are a way of organizing people to accomplish a discrete task and are relatively easy to create and deprecate when inactive.
```
Working groups are a way of organizing people to accomplish a discrete task, and are relatively easy to create and deprecate when inactive.

zacharysarah commented on _data/glossary/wg.yaml:
```diff
@@ -0,0 +1,9 @@
+  Facilitates the discussion and/or implementation of a short-lived, narrow, or decoupled project for a committee, SIG, or cross-SIG effort.
+long-description: |
+  They are a way of organizing people to accomplish a discrete task and are relatively easy to create and deprecate when inactive.
+  For more information, see the [kubernetes/community](https://github.com/kubernetes/community) repo and the current list of [SIGs and Working Groups](https://github.com/kubernetes/community/blob/master/sig-list.md).
```
working groups

abiogenesis-now commented on _data/glossary/deployment.yaml:
```diff
@@ -0,0 +1,10 @@
+short-description: |
+  An API object that manages a replicated application.
+long-description: |
+  Each replica is represented by a Pod, and the Pods are distributed among the Nodes of a cluster.
```
:+1: will stick to capitalize standard K8s objects only

zacharysarah replied:
@abiogenesis-now Thanks, that was what I was trying to convey. 😸 👍

abiogenesis-now commented on _data/glossary/developer.yaml:
```diff
@@ -0,0 +1,15 @@
+  May refer to&#58; *Application Developer*, *Code Contributor*, or *Platform Developer*.
+long-description: |
+  This overloaded term may have different meanings depending on the context. Respectively, it could mean&#58;
+
```
I think it's required for formatting, leaving for now.

abiogenesis-now commented on _data/glossary/downstream.yaml:
```diff
@@ -0,0 +1,12 @@
+short-description: |
+  May refer to: code in the Kubernetes ecosystem that depends upon the core Kubernetes codebase or a forked repo.
+long-description: |
+  In the Kubernetes Community, conversations often use *downstream* to mean the ecosystem, code, or third-party tools that rely on the core Kubernetes codebase. For example, a new feature in Kubernetes may be adopted by applications *downstream* to improve their functionality.
```
@chenopis ++ !

abiogenesis-now commented:
Addressed most of your comments @zacharysarah and added some more cross-linking. Also am now using `long-description: >` rather than `long-description: |` in some of the YAML files because this allows you to break up your text semantically (without actually resulting in rendered linebreaks).
